internal/multistorage/cache/alive_check.go

```diff
@@ -52,11 +56,13 @@ func checkForAlive(timeout time.Duration, storages ...NamedFolder) map[key]bool
 	return results
 }
 
-func checkStorage(ctx context.Context, folder NamedFolder) error {
-	// Currently, we use the simple ListFolder() call to check if the storage is up and reachable
+func checkStorage(ctx context.Context, size uint, folder NamedFolder) error {
 	errCh := make(chan error, 1)
+	// todo simplify code using PutObjectWithContext after merge https://github.com/wal-g/wal-g/pull/1546
 	go func() {
-		_, _, err := folder.ListFolder()
+		r := rand.New(rand.NewSource(1))
+		lr := io.LimitReader(r, int64(size))
+		err := folder.PutObject(checkObjectName, lr)
```

Comment:
Don't we want to check storages by both putting an object and listing the folder?